src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -110,9 +116,11 @@ def create_project_cohort(worker_number: int, last_processed_project_id: int | N
     total_worker_count = options.get("similarity.backfill_total_worker_count")
     cohort_size = options.get("similarity.backfill_project_cohort_size")
 
+    query = Project.objects.filter(project_id_filter)
+    if skip_processed_projects:
+        query = query.exclude(projectoption__key=PROJECT_BACKFILL_COMPLETED)
```

Comment:
I think we can remove these [lines](https://github.com/getsentry/sentry/blob/abe3cd7014a8f81cae6ad29c76501ac14106012b/src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py#L138-L151) as it'll be redundant to this change

Reply:
I have added the tests. I will look into this comment now.